api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -171,6 +171,21 @@ message HttpConnectionManager {
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
   uint32 xff_num_trusted_hops = 19;
 
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header. This may be used in
+  // conjunction with HTTP filters that explicitly manipulate XFF after the HTTP connection manager
+  // has mutated the request headers.
+  bool skip_xff_append = 21;
+
+  // Do not consider requests to ever be of internal provenance.
+  bool external_only = 22;
+
+  // Do not add any additional *x-envoy-* headers to requests or responses.
+  bool suppress_envoy_headers = 23;
```

Comment:
I think you probably need this config option on the router also? Or is your plan to make at a filter callback value that the router can lookup and pivot on?